src/activities/ActivityUsageEntity.js

```diff
@@ -227,10 +277,47 @@ export class ActivityUsageEntity extends Entity {
 	 * @returns {string} End date of the activity usage
 	 */
 	endDate() {
-		const endDate = this._getDateSubEntity(Classes.dates.endDate);
+		const availabilityEndDate = this._getAvailabilityEndDateEntity();
+
+		if (availabilityEndDate) {
+			return availabilityEndDate;
```

Comment:
Should this be returning `return availabilityEndDate.properties.dateTime.date;`?